frame/identity/src/lib.rs

```diff
@@ -872,6 +877,36 @@ decl_module! {
 
 			Self::deposit_event(RawEvent::IdentityKilled(target, deposit));
 		}
+
+		fn on_initialize() {
+			if !IsUpgraded::get() {
+				IsUpgraded::put(true);
+				Self::do_upgrade();
+			}
+		}
+	}
+}
+
+// Migration code to update storage name
+use frame_support::storage::migration::{put_storage_value, take_storage_value, StorageIterator};
+impl<T: Trait> Module<T> {
+	pub fn do_upgrade() {
+		sp_runtime::print("Migrating Identity.");
+
+		for (hash, identity_of) in StorageIterator::<Option<Registration<BalanceOf<T>>>>::new(b"Sudo", b"IdentityOf").drain() {
+			put_storage_value(b"Identity", b"IdentityOf", &hash, identity_of);
```

Comment:
note that we could make use of `<IdentityOf<T>>::storage_prefix()` instead of `b"IdentityOf"` to avoid typos but I'm ok with this if tested.